recipes/ms-gsl/all/conanfile.py

```diff
@@ -70,6 +70,9 @@ def package_info(self):
         self.cpp_info.names["cmake_find_package_multi"] = "Microsoft.GSL"
         self.cpp_info.components["_ms-gsl"].names["cmake_find_package"] = "GSL"
         self.cpp_info.components["_ms-gsl"].names["cmake_find_package_multi"] = "GSL"
+        self.cpp_info.set_property("cmake_file_name", "Microsoft.GSL")
+        self.cpp_info.set_property("cmake_target_namespace", "Microsoft.GSL")
+        self.cpp_info.components["_ms-gsl"].set_property("cmake_target_name", "GSL")
```

Comment:
It's confusing to keep the old mechanism. I would remove it and bump `required_conan_version`

Reply:
I can do that.

Reply:
Also you don't need anymore to use components I think:

```python
    def package_info(self):
        self.cpp_info.set_property("cmake_file_name", "Microsoft.GSL")
        self.cpp_info.set_property("cmake_target_namespace", "Microsoft.GSL")
        self.cpp_info.set_property("cmake_target_name", "GSL")
        if tools.Version(self.version) < "3.0.0":
            self.cpp_info.defines = [
                self._contract_map[str(self.options.on_contract_violation)]
            ]
```

Reply:
Doesn't work with the test package. Do I have to modify the test package to use the `CMakeDeps` generator. Would this change not be backwards compatible with the other generators?

Reply:
It's important to still properly support cmake_find_package and cmake_find_package_multi in CCI recipes, so forget my last suggestion.

Reply:
Okay, I've committed and reverted this change so its in the history, but for now I'm leaving the changes as they were with the original functions for the other generators. The component part `self.cpp_info.components["_ms-gsl"].set_property("cmake_target_name", "GSL")` is also necessary, or things won't build for the test package.

Reply:
Ok so I've tested with cmake_find_package, cmake_find_package_multi and CMakeDeps, here is a good trade-off 

```python
required_conan_version = ">=1.36.0"

(...)

    def package_info(self):
        self.cpp_info.set_property("cmake_file_name", "Microsoft.GSL")
        self.cpp_info.set_property("cmake_target_name", "Microsoft.GSL")
        self.cpp_info.components["_ms-gsl"].set_property("cmake_target_name", "GSL")
        if tools.Version(self.version) < "3.0.0":
            self.cpp_info.components["_ms-gsl"].defines = [
                self._contract_map[str(self.options.on_contract_violation)]
            ]
```

Reply:
Works for me! Done.